src/docfx/lib/HtmlUtility.cs

```diff
@@ -163,27 +164,24 @@ public static string TransformLinks(string html, Func<string, string> transform)
             return result.ToString();
         }
 
-        public static HtmlNode ExtractTitle(HtmlNode node)
+        public static HtmlNode RemoveTitle(HtmlNode node)
         {
+            var previousVisableNode = false;
             foreach (var child in node.ChildNodes)
             {
-                if (child.NodeType == HtmlNodeType.Comment)
+                if (child.NodeType != HtmlNodeType.Comment &&
+                    !(child.NodeType == HtmlNodeType.Text && string.IsNullOrWhiteSpace(child.OuterHtml)))
                 {
-                    continue;
-                }
+                    if (child.NodeType == HtmlNodeType.Element && (child.Name == "h1" || child.Name == "h2" || child.Name == "h3"))
```

Comment:
Should check if all the nodes before h1 is visible instead of just the previous one, e.g.
```
asdf

<!---comment-->
h1

```

Reply:
just check any one is visible is enough, no need to check all are invisible